Cookbook/Technical-Documents/Development-Process.md

```diff
@@ -0,0 +1,61 @@
+- The ticket is moved to `In progress` (this might be automatically done by the bot when branch published, if configured)
+
+## Get started with development 🛠
+
+- Create a branch where this work will live. Please follow this format: `[NAME]/[TICKET]-[DESCRIPTION (optional)]` (E.g. `john/TCKT-123-fix-bug`)
+- For new features, add a [local feature switch](https://github.com/babylonhealth/ios-playbook/blob/master/Cookbook/Technical-Documents/FeatureSwitches.md) to avoid getting WIP code into the release. Add the name of this flag (and additional instructions) to the ticket to help QA test it
+- Implement all features, changes to satisfy the acceptance criteria defined in the ticket
+- If it introduces a new component, add it to the `GalleryApp` and make sure it's included in the Design System
+
+## Testing 🧪 <!-- TODO: [CNSMR-3195] link to testing article -->
+
+- Make sure:
+  - New components, screens have snapshot tests
+  - Logic, ViewModels, SDK changes are unit tested
+  - Automation tests have been updated or added if the feature will be in the next release
+- Build `GalleryApp` and `Babylon` before merging to avoid checks failing
+
+## Creating the PR 📝
+
+- Fill in the PR template and set the title using the following format: `[TCKT-123] Short summary of work`
+- [Add labels](https://github.com/babylonhealth/ios-playbook/blob/master/Cookbook/Technical-Documents/LabelsInPRs.md)
+- Assign the PR to the engineer(s)
+- Move the ticket to `Peer review` on Jira (or it'll be automatically moved if configured)
+- If you'd like feedback before the work on the ticket is done, open a draft PR so your peers can leave comments
+- To run checks before merging, use the comment `@ios-bot-babylon test_pr` ([more info](https://github.com/babylonhealth/ios-playbook/blob/master/Cookbook/Technical-Documents/SlackCIIntegration.md))
+
+_Note: Sometimes Pull Assigner is not triggered when a new PR is opened. In that case, re-request `ios-pullassigner` to add reviewers from the team._
+
+## Merging PR 🚦
+
+- Once a PR has 2 approvals and no outstanding comments/changes requested, the `Merge` label can be added to begin the merging process
+- Our [bot](https://github.com/babylonhealth/Wall-E) will add it to the merge queue and integrate it into the rest of the project
+- Once it reaches the front, it's updated with the target branch and CircleCI runs all necessary tests to make sure nothing's failing. _Note: there's no need to trigger checks manually, if there are other PRs in the queue._ <!-- TODO: [CNSMR-3231] link to Wall-E article -->
+- If the required checks are passing, the PR is merged and the branch deleted.
+- The ticket is moved to `Awaiting build` on Jira by our [bot](https://github.com/babylonhealth/Stevenson). It will be updated to `Ready for QA` when the next App Center build is created. (if configured, otherwise this should be done manually)
+
```

Comment:
```suggestion
- If the required checks fail you will have to analyse what is the reason for the failure, fix it and add the `Merge` label again.
```

Reply:
Checking the "required checks" part, there was a recent change in Wall-E that might have changed this

Reply:
All checks are needed, not just required ones
```suggestion
- If the checks fail, you will have to analyse what is the reason for the failure, fix it, and add the `Merge` label again
```